### PR TITLE
[4.1][regression] Correctly handle save2copy for menus

### PR DESCRIPTION
**Pull request by zero-24**
Pull Request for Issue #37812

### Summary of Changes

Restore the existing behavior before the regression caused by https://github.com/joomla/joomla-cms/pull/37395 as well as fix the Issue from the original report

### Testing Instructions

- setup a 4.1.2 site
- create 2 menus
- create one menu items for the first menu
- open that menu item in edit mode
- change the menu setting to the seccond menu
- Use the "save2copy" feature.
- the change to the menu setting is ignored
- update to 4.1.3
- repeat the steps from above
- it works
- but try it without changing the assigned menu
- it fails as the alias exists already
- apply the patch
- confirm that both cases still work as expected
- try to only change the menu item title
- save2copy
- confirm that the alias number is not just increased but followed the change to the menu title
- try to only change the alias but keep the title
- save2copy
- confirm that the alias has been updated but not the title.

### Actual result BEFORE applying this Pull Request

Issues of "alias exists already" while with save2copy we have an "increment the ID" feature

### Expected result AFTER applying this Pull Request

The alias will be updated as expected any by using save2copy you should not get alias dubliceted issues.

### Documentation Changes Required

n.a

### Mentions

@toivo 
@bembelimen 

Involved within the PR which caused the regression:
@anmode
@chmst
@richard67
@vorayash
@Quy

### CODE REVIEW

**Comment by toivo** [comment]
I have tested this item :white_check_mark: successfully on 096a742d0762a9e31940a12ae06099c634da7472<br /><br />Tested successfully in 4.1.2 upgraded to 4.1.3 and patched, using PHP 8.0.15.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37813">issues.joomla.org/tracker/joomla-cms/37813</a>.</sub>

**Comment by Webdongle** [comment]
I have tested this item :white_check_mark: successfully on 096a742d0762a9e31940a12ae06099c634da7472<br /><br />Full success<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37813">issues.joomla.org/tracker/joomla-cms/37813</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37813">issues.joomla.org/tracker/joomla-cms/37813</a>.</sub>

File: administrator/components/com_menus/src/Model/ItemModel.php [hunk 1]
```diff
@@ -1512,13 +1512,24 @@ public function save($data)
 			return false;
 		}
 
-		// Alter the title & alias for save as copy.  Also, unset the home record.
-		if (!$isNew && $data['id'] == 0)
+		// Alter the title & alias for save2copy when required. Also, unset the home record.
+		if (Factory::getApplication()->input->get('task') === 'save2copy' && $data['id'] === 0)
```
Comment (laoneo) [inline administrator/components/com_menus/src/Model/ItemModel.php:1516]: This code here should be moved to a controller. The model is the wrong place. Please change that.

**Review by laoneo — CHANGES_REQUESTED** [review]

File: administrator/components/com_menus/src/Model/ItemModel.php (hunk 1 above)
Comment (bembelimen) [reply administrator/components/com_menus/src/Model/ItemModel.php:1516]: It's like currently everywhere in Joomla!, so I'm ok to merge (beside that I think, we should improve this code everywhere.)

**Comment by bembelimen** [comment]
Thx

**Pull request merged**